tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,210 @@
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude any artists marked as 'animated'
+- save a copy of the RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.  One consequence of this procedure is that your animated
+artists are always drawn at a higher z-order than the static artists.
+
+Not all backends support blitting.  You can check if a given canvas does via
+the `.FigureCanvasBase.supports_blit` property.
+
+Minimal example
+---------------
+
+We can use the `.FigureCanvasAgg` methods
+`~.FigureCanvasAgg.copy_from_bbox` and
+`~.FigureCanvasAgg.restore_region` in conjunction with setting
+``animated=True`` on our artist to implement a minimal example that
+uses blitting to accelerate rendering
+
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2 * np.pi, 100)
+
+fig, ax = plt.subplots()
+# animated=True tells matplotlib to only draw the artist when we
+# explicitly request it
+(ln,) = ax.plot(x, np.sin(x), animated=True)
+
+# make sure the window is raised, but the script keeps going
+plt.show(block=False)
+# stop to admire our empty window axes and ensure it is rendered at
+# least once
+plt.pause(0.1)
+
+# get copy of entire figure (everything inside fig.bbox) sans animated artist
+bg = fig.canvas.copy_from_bbox(fig.bbox)
+# draw the animated artist
+ax.draw_artist(ln)
+# show the result to the screen
+fig.canvas.blit(fig.bbox)
+
+for j in range(100):
+    # put the un-changed background back
+    fig.canvas.restore_region(bg)
+    # update the artist.
+    ln.set_ydata(np.sin(x + (j / 100) * np.pi))
+    # re-render the artist
+    ax.draw_artist(ln)
+    # copy the result to the screen
+    fig.canvas.blit(fig.bbox)
+    # make sure the GUI has flushed any pending events
+    fig.canvas.flush_events()
+    # you can put a pause in if you want to slow things down
+    # plt.pause(.1)
+
+###############################################################################
+# This example works and shows a simple animation, however because we
+# are only grabbing the background once, if the size of the figure in
+# pixels changes (due to either the size or dpi of the figure
+# changing) , the background will be invalid and result in incorrect
+# (but sometimes cool looking!) images.  There is also a global
+# variable and a fair amount of boiler plate which suggests we should
+# wrap this in a class.
+#
+# Class-based example
+# -------------------
+#
+# We can use a class to encapsulate the boilerplate logic and state of
+# restoring the background, drawing the artists, and then blitting the
+# result to the screen.  Additionally, we can use the ``'draw_event'``
+# callback to capture a new background whenever a full re-draw
+# happens.
```

Comment:
Maybe end this with "this solves the issues with screen resizing mentioned above"? I think that's not clear (I initially even left a comment reminding you to mention that this still does not fix that very problem before I tested it and realized it does).